PlagiarismPlugin.inc.php

```diff
@@ -43,215 +132,959 @@ public function getDescription() {
+	/**
+	 * Stamp the iThenticate EULA to the submitting user
+	 * 
+	 * @param Context 		$context
+	 * @param Submission 	$submission
+	 * @param User|null 	$user
+	 * 
+	 * @return bool
+	 */
+	public function stampEulaToSubmittingUser($context, $submission, $user = null) {
+		$request = Application::get()->getRequest();
+		$user ??= $request->getUser();
+
+		$submissionEulaVersion = $submission->getData('ithenticateEulaVersion');
+
+		// If submission EULA version has already been stamped to user
+		// no need to do the confirmation and stamping again
+		if ($user->getData('ithenticateEulaVersion') === $submissionEulaVersion) {
+			return false;
+		}
+
+		$ithenticate = $this->initIthenticate(...$this->getServiceAccess($context)); /** @var IThenticate $ithenticate */
+		$ithenticate->setApplicableEulaVersion($submissionEulaVersion);
+		
+		// Check if user has ever already accepted this EULA version and if so, stamp it to user
+		// Or, try to confirm the EULA for user and upon succeeding, stamp it to user
+		if ($ithenticate->verifyUserEulaAcceptance($user, $submissionEulaVersion) ||
+			$ithenticate->confirmEula($user, $context)) {
+			$this->stampEulaVersionToUser($user, $submissionEulaVersion);
+			return true;
+		}
+
+		return false;
 	}
 
-	public function createFolder($folder_name, $folder_description, $group_id, $exclude_quotes) {
-		error_log("Creating folder:\n\t$folder_name\n\t$folder_description\n\t$group_id\n\t$exclude_quotes");
+	/**
+	 * Create a new submission at iThenticate service's end
+	 * 
+	 * @param Request 						$request
+	 * @param User 							$user
+	 * @param Submission 					$submission
+	 * @param SubmissionFile 				$submissionFile
+	 * @param IThenticate|TestIThenticate 	$ithenticate
+	 * 
+	 * @return bool
+	 */
+	public function createNewSubmission($request, $user, $submission, $submissionFile, $ithenticate) {
+		$context = $request->getContext();
+		$publication = $submission->getCurrentPublication();
+		$author = $publication->getPrimaryAuthor();
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /** @var SubmissionFileDAO $submissionFileDao */
+
+		$submissionUuid = $ithenticate->createSubmission(
+			$request->getSite(),
+			$submission,
+			$user,
+			$author,
+			static::SUBMISSION_AUTOR_ITHENTICATE_DEFAULT_PERMISSION,
+			$this->getSubmitterPermission($context, $user)
+		);
+
+		if (!$submissionUuid) {
+			$this->sendErrorMessage("Could not create the submission at iThenticate for file id {$submissionFile->getId()}", $submission->getId());
+			return false;
+		}
+
+		$pkpFileService = Services::get('file'); /** @var \PKP\Services\PKPFileService $pkpFileService */
+		$file = $pkpFileService->get($submissionFile->getData('fileId'));
+
+		if (in_array($file->mimetype, $this->uploadRestrictedArchiveMimeTypes)) {
+			error_log("plagiarism check for file mime type : {$file->mimetype} with submission file id : {$submissionFile->getId()} and submission id : {$submission->getId()} is not allowed");
```

Comment:
I think this will just clutter the logs. I'd suggest removing the `error_log` call.